repository/src/main/java/ru/karalionak/rest/dao/impl/CertificateDaoImpl.java

```diff
@@ -4,50 +4,50 @@
 import org.springframework.stereotype.Repository;
 import ru.karalionak.rest.dao.CertificateDao;
 import ru.karalionak.rest.entity.Certificate;
+import ru.karalionak.rest.mapper.ParametersMapper;
 import ru.karalionak.rest.query.OrderedViewQuery;
-import ru.karalionak.rest.query.impl.QueryHelperImpl;
+import ru.karalionak.rest.query.QueryHelper;
 
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
 
 @Repository
 public class CertificateDaoImpl extends AbstractDao<Certificate> implements CertificateDao {
     private static final String CREATE_CERTIFICATE =
-            "INSERT INTO gift_certificates(certificate_name, description, price, duration) VALUES(?, ?, ?, ?)";
+            "INSERT INTO gift_certificates(certificate_name, description, price, duration) " +
+                    "VALUES(:name, :description, :price, :duration) RETURNING id";
     private static final String CREATE_REFERENCE_BETWEEN_CERTIFICATE_TAG =
             "INSERT INTO gift_certificate_tag(certificate_id, tag_id) VALUES(?,?)";
-    private static final String FIND_CERTIFICATE_BY_TAG_ID =
-            "SELECT * FROM gift_certificates JOIN gift_certificate_tag gct on gift_certificates.id = gct.certificate_id " +
-                    "WHERE gct.tag_id=?";
     private final static String TABLE_NAME = "gift_certificates";
-    private final QueryHelperImpl queryHelper;
     private final ResultSetExtractor<List<Certificate>> certificateExtractor;
 
-    public CertificateDaoImpl(JdbcTemplate jdbcTemplate, QueryHelperImpl queryHelper,
-                              RowMapper<Certificate> certificateRowMapper,
-                              ResultSetExtractor<List<Certificate>> certificateExtractor) {
-        super(jdbcTemplate, certificateRowMapper, TABLE_NAME);
-        this.queryHelper = queryHelper;
+    public CertificateDaoImpl(NamedParameterJdbcTemplate namedJdbcTemplate, ParametersMapper<Certificate> parametersMapper,
+                              JdbcTemplate jdbcTemplate, RowMapper<Certificate> certificateRowMapper,
+                              QueryHelper queryHelper, ResultSetExtractor<List<Certificate>> certificateExtractor) {
+        super(namedJdbcTemplate, parametersMapper, jdbcTemplate, certificateRowMapper, TABLE_NAME, queryHelper);
         this.certificateExtractor = certificateExtractor;
     }
 
     @Override
-    public void create(Certificate certificate) {
-        jdbcTemplate.update(CREATE_CERTIFICATE, certificate.getName(), certificate.getDescription(),
-                certificate.getPrice(), certificate.getDuration());
+    public long create(Certificate certificate) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        namedJdbcTemplate.update(CREATE_CERTIFICATE, parametersMapper.map(certificate), keyHolder);
+        return keyHolder.getKey().longValue();
```

Comment:
Intelij indicates its as problem. Try to avoid such situations. One of the solutions:
`Objects.requireNonNull(keyHolder.getKey(), "Database did not return primary key after certificate creation")
                .longValue();`